cw-dex/tests/astroport_tests.rs

```diff
@@ -0,0 +1,306 @@
+use apollo_utils::coins::coin_from_str;
+use apollo_utils::submessages::{find_event, parse_attribute_value};
+use astroport::factory::PairType;
+use cosmwasm_std::{Coin, Decimal, SubMsgResponse, Uint128};
+use cw_asset::{Asset, AssetInfo, AssetInfoBase, AssetList};
+use cw_dex::astroport::msg::PoolResponse;
+use cw_dex::implementations::astroport::msg::PairQueryMsg;
+use cw_dex_test_contract::msg::{AstroportExecuteMsg, ExecuteMsg, QueryMsg};
+use cw_dex_test_helpers::astroport::setup_pool_and_test_contract;
+use cw_dex_test_helpers::{
+    cw20_balance_query, cw20_transfer, provide_liquidity, query_asset_balance,
+};
+use osmosis_testing::cosmrs::proto::cosmwasm::wasm::v1::MsgExecuteContractResponse;
+use osmosis_testing::{
+    Account, ExecuteResponse, Module, OsmosisTestApp, Runner, RunnerResult, SigningAccount, Wasm,
+};
+use test_case::test_case;
+
+const TEST_CONTRACT_WASM_FILE_PATH: &str =
+    "../target/wasm32-unknown-unknown/release/astroport_test_contract.wasm";
+
+fn setup_pool_and_contract(
+    pool_type: PairType,
+    initial_liquidity: Vec<(&str, u64)>,
+) -> RunnerResult<(
+    OsmosisTestApp,
+    Vec<SigningAccount>,
+    String,
+    String,
+    String,
+    AssetList,
+)> {
+    setup_pool_and_test_contract(
+        pool_type,
+        initial_liquidity,
+        2,
+        TEST_CONTRACT_WASM_FILE_PATH,
+    )
+}
+
+#[test_case(PairType::Xyk { }, vec![("uluna",1_000_000), ("astro", 1_000_000)]; "provide_liquidity: native-cw20")]
+#[test_case(PairType::Xyk { }, vec![("apollo",1_000_000), ("astro", 1_000_000)]; "provide_liquidity: cw20-cw20")]
+#[test_case(PairType::Stable { }, vec![("uluna",1_000_000), ("astro", 1_000_000)]; "provide_liquidity: stableswap native-cw20")]
+#[test_case(PairType::Stable { }, vec![("apollo",1_000_000), ("astro", 1_000_000)]; "provide_liquidity: stableswap cw20-cw20")]
+pub fn test_provide_liquidity(pool_type: PairType, initial_liquidity: Vec<(&str, u64)>) {
+    let (runner, accs, lp_token_addr, _pair_addr, contract_addr, asset_list) =
+        setup_pool_and_contract(pool_type, initial_liquidity).unwrap();
+    let admin = &accs[0];
+
+    // Check contract's LP token balance before providing liquidity
+    let lp_token_before =
+        cw20_balance_query(&runner, lp_token_addr.clone(), contract_addr.clone()).unwrap();
+    assert_eq!(lp_token_before, Uint128::zero());
+
+    // Provide liquidity
+    provide_liquidity(
+        &runner,
+        contract_addr.clone(),
+        asset_list.clone(),
+        Uint128::one(),
+        admin,
+    );
+
+    // Query LP token balance after
+    let lp_token_after = cw20_balance_query(&runner, lp_token_addr, contract_addr.clone()).unwrap();
+    assert_ne!(lp_token_after, Uint128::zero());
```

Comment:
can we assert how many tokens we should get?

Reply:
Maybe, but the number depends on the version of Astroport contracts used. I'd say this is fine for now.